File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/server/httphandlers/InfoGet.java
package smarthome.raspberry.server.httphandlers;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import fi.iki.elonen.NanoHTTPD;
import smarthome.raspberry.model.SmartHomeRepository;

import static fi.iki.elonen.NanoHTTPD.MIME_PLAINTEXT;

public class InfoGet extends BaseRequestHandler {
    private Gson gson;

    public InfoGet() {
        gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create();
    }

    @Override
    public NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
        String serializedHomeState = gson.toJson(SmartHomeRepository.getInstance());
        return new NanoHTTPD.Response(NanoHTTPD.Response.Status.OK, MIME_PLAINTEXT, serializedHomeState);
    }
}

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/server/httphandlers/InfoGet.kt
package smarthome.raspberry.server.httphandlers

import com.google.gson.GsonBuilder
import fi.iki.elonen.NanoHTTPD
import fi.iki.elonen.NanoHTTPD.MIME_PLAINTEXT
import smarthome.raspberry.model.SmartHomeRepository

class InfoGet : BaseRequestHandler() {
    private val gson = GsonBuilder().excludeFieldsWithoutExposeAnnotation().create()

    override fun serve(session: NanoHTTPD.IHTTPSession): NanoHTTPD.Response {
        val serializedHomeState = gson.toJson(SmartHomeRepository)
        return NanoHTTPD.Response(NanoHTTPD.Response.Status.OK, MIME_PLAINTEXT, serializedHomeState)
    }
}
